src/Display/Modal/Modal.tsx

```diff
@@ -26,9 +27,18 @@ class Modal extends React.Component<Props, State> {
     this.modalContentAreaRef = React.createRef();
   }
 
-  componentDidMount() {
+  guardedOnClose() {
     const { onClose } = this.props;
-    document.addEventListener('keydown', escEvent(onClose), false);
+    if (this.state.isOpen) onClose();
+  }
+
+  registerEventEscapeListener() {
+    this.escEvent = escEvent(this.guardedOnClose.bind(this));
+    document.addEventListener('keydown', this.escEvent, false);
+  }
+
+  componentDidMount() {
+    this.registerEventEscapeListener();
```

Comment:
maybe we can `addEventListener` only when the modal is visible, then we don't need the `guardedOnClose`
 ```
componentDidMount() {
  if (!this.props.isVisible) return;
  this.registerEventEscapeListener();
}
```

Reply:
and suggested changes on registerEventEscapeListener and removeEscapeEventListener:
```
onModalOpen() {
  this.escEvent = escEvent(this.guardedOnClose.bind(this));
  document.addEventListener('keydown', this.escEvent, false);
  document.body.style.overflow = 'hidden';
}

onModalClose() {
  document.removeEventListener('keydown', this.escEvent, false);
  document.body.removeAttribute('style');
}
```

Reply:
the state `isOpen` seems to be a duplicate of the prop `isVisible`, can we remove it?

Reply:
> the state isOpen seems to be a duplicate of the prop isVisible, can we remove it?
I tried this but it increased the scope of this so I decided not to do it. I kinda need to merge this one soon.

> so we can remove `getDerivedStateFromProps`
Same as above. `getDerivedStateFromProps` is needed for the state.

> maybe we can `addEventListener` only when the modal is visible
Thanks for the suggestion, updated! Much more readable this way.